FetchMigration/python/migration_monitor.py

```diff
@@ -65,32 +85,82 @@ def check_if_complete(doc_count: Optional[int], in_flight: Optional[int], no_par
+    terminal: bool = False
+    # If the API call fails, the response is empty
+    metrics = fetch_prometheus_metrics(endpoint)
+    if metrics is not None:
+        # Reset API failure counter
+        internal_metrics.seq_metric_api_fail = 0
+        success_docs = get_metric_value(metrics, __DOC_SUCCESS_METRIC)
+        rec_in_flight = get_metric_value(metrics, __RECORDS_IN_FLIGHT_METRIC)
+        no_partitions_count = get_metric_value(metrics, __NO_PARTITIONS_METRIC)
+        if success_docs is not None:  # pragma no cover
+            # Reset value failure counter
+            internal_metrics.seq_metric_value_fail = 0
+            completion_percentage: int = math.floor((success_docs * 100) / target_doc_count)
+            progress_message: str = "Completed " + str(success_docs) + \
+                                    " docs ( " + str(completion_percentage) + "% )"
+            logging.info(progress_message)
+        else:
+            internal_metrics.seq_metric_value_fail += 1
+            logging.warning("Could not fetch progress stats from Data Prepper response, " +
+                            "will retry on next polling cycle...")
+        terminal = check_if_complete(success_docs, rec_in_flight, no_partitions_count,
+                                     internal_metrics.prev_no_partitions_count, target_doc_count)
+        if not terminal:
+            # Save no_partitions_count
+            internal_metrics.prev_no_partitions_count = no_partitions_count
+    else:
+        internal_metrics.seq_metric_api_fail += 1
+        logging.warning("Data Prepper metrics API call failed, will retry on next polling cycle...")
+    # TODO - Handle idle non-terminal pipeline
+    return terminal, internal_metrics
+
+
+def monitor_local(args: MigrationMonitorParams, dp_process: Popen, poll_interval_seconds: int = 30) -> Optional[int]:
+    endpoint = EndpointInfo(args.data_prepper_endpoint)
+    target_doc_count: int = args.target_count
+    # Counters to track idle pipeline and failures
+    internal_metrics = MigrationMonitorCounters()
+    # Track whether the migration has reached a terminal state
+    terminal = False
+    logging.info("Starting migration monitor until target doc count: " + str(target_doc_count))
+    # Sets returncode. A value of None means the subprocess has not yet terminated
+    dp_process.poll()
+    while dp_process.returncode is None and not terminal:
+        try:
+            dp_process.wait(timeout=poll_interval_seconds)
+        except subprocess.TimeoutExpired:
+            pass
+        if dp_process.returncode is None:
+            terminal, internal_metrics = check_and_log_progress(endpoint, internal_metrics, target_doc_count)
+    # Loop terminated
+    if not terminal:
+        logging.error("Migration did not complete, process exited with code: " + str(dp_process.returncode))
+        # TODO - Implement rollback
```

Comment:
You should tell the user that there's no rollback and that they should consult documentation on how to recover.

Reply:
Added error logging